resources/js/services/requestFunction.js

```diff
@@ -0,0 +1,19 @@
+ export function createOptions(method, data = null) {
+    return {
+        method: method,
+        headers: {
+            'Content-Type': 'application.json',
+            'X-CSRF-Token': _token,
+        },
+        body: JSON.stringify(data)
+    }
+}
+
+ export function request(url, options = null) {
+
+   return  fetch(url, options).then(response =>  response.json())
+        // .then(data => {setter(data)})
+        // .catch(error => {
+        //     console.error(error);
+        // });
```

Comment:
Функция получилась не универсальная. Сейчас она ожидает что ответ вернется в формате `json`, но это может быть не так. Стоит это держать в голове при использовании этой функции. 

```suggestion
   return  fetch(url, options).then(response =>  response.json())
```